src/main/java/core/basesyntax/handler/impl/BalanceTransactionHandler.java

```diff
@@ -0,0 +1,12 @@
+package core.basesyntax.handler.impl;
+
+import core.basesyntax.handler.TransactionHandler;
+import java.util.Map;
+
+public class BalanceTransactionHandler implements TransactionHandler {
+    @Override
+    public void handleTransaction(Map<String, Integer> mapFruitQuantity, String key, int value) {
```

Comment:
Follow the encapsulation principle. The method `handleTransaction` should not expose the `mapFruitQuantity` directly. Consider providing a more encapsulated way to handle transactions.